### PR TITLE
Encode nil as null

### LINKED ISSUE #91: Encoder / Decoder - ability to translate null -> nil for Elixir clients
Thanks for the great project! 

I am interested in implementing this myself, but I thought I'd raise it as an issue if someone got to it first. For Elixir clients of `erlavro`, `nil` needs to be swapped with `:null` on the encoding side, and on the decoding side, `:null` should be swapped with `nil`.

Since this might not be desirable for all consumers, it seems that this could just be an option passed to `:avro.make_simple_encoder/2` and `:avro.make_simple_decoder/2`

### PR DESCRIPTION
**Pull request by davydog187**
Fixes the encoding part of #91. I'll open another for the decoding

This is admittedly my first Erlang code ever, so I may have botched something.

### CODE REVIEW

**Comment by davydog187** [comment]
@zmstone do you have any feedback on this PR?

**Comment by zmstone** [comment]
Look good to me.
ping @mikpe @qzhuyan @k32 for approval and merge.

**Comment by zmstone** [comment]
oh, maybe update changelog.md and bump a minor.

**Comment by davydog187** [comment]
@zmstone I bumped the patch version and added to changelog. 

CC: @mikpe @qzhuyan @k32 for review/approval

File: src/avro_primitive.erl [hunk 1]
```diff
@@ -89,6 +89,8 @@ string_type()  -> type(?AVRO_STRING, []).
 -spec cast(avro_type(), term()) -> {ok, avro_value()} | {error, term()}.
 cast(Type, null) when ?IS_NULL_TYPE(Type) ->
   {ok, ?AVRO_VALUE(Type, null)};
+cast(Type, nil) when ?IS_NULL_TYPE(Type) ->
```
[the remark this answers is not in the document]
Comment (davydog187) [reply src/avro_primitive.erl]: Done

**Comment by davydog187** [comment]
@mikpe looks like you need to merge for me :)

Would it be possible to do a release as well?

**Comment by mikpe** [comment]
Merged and tagged.

**Pull request merged**